bgpd/bgp_routemap.c

```diff
@@ -5501,6 +5579,51 @@ DEFUN_YANG (no_match_ecommunity,
 }
 
 
+DEFPY_YANG (set_ecommunity_delete,
+	    set_ecommunity_delete_cmd,
+            "set extended-comm-list <(1-99)|(100-500)|EXTCOMMUNITY_LIST_NAME> delete",
+	    SET_STR
+	    "set BGP extended community list (for deletion)\n"
+	    "Extended community-list number (standard)\n"
+	    "Extended community-list number (expanded)\n"
+	    "Extended community-list name\n"
+            "Delete matching extended communities\n")
+{
+	const char *xpath =
+		"./set-action[action='frr-bgp-route-map:extended-comm-list-delete']";
+	char xpath_value[XPATH_MAXLEN];
+	int idx_comm_list = 2;
+
+	nb_cli_enqueue_change(vty, xpath, NB_OP_CREATE, NULL);
+
+	snprintf(xpath_value, sizeof(xpath_value),
+		 "%s/rmap-set-action/frr-bgp-route-map:comm-list-name",
+		 xpath);
+	nb_cli_enqueue_change(vty, xpath_value, NB_OP_MODIFY,
+			argv[idx_comm_list]->arg);
+	return nb_cli_apply_changes(vty, NULL);
+}
+
+
+DEFPY_YANG (no_set_ecommunity_delete,
+	    no_set_ecommunity_delete_cmd,
+            "no set extended-comm-list <(1-99)|(100-500)|EXTCOMMUNITY_LIST_NAME> delete",
```

Comment:
We should allow an operator to do `no set extended-comm-list` without an additional arguments also.